### PR TITLE
feat(GridFormSectionTitle): Add Text variant to GridFormSection

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview
Adds the variant Text prop to GridFormSection to make GridFormTitle configurable.

### PR Checklist

- [x] Related to designs: [link](https://www.figma.com/file/PG0kVYDIuLk3hE566beZUV/Profiles-(Concept-Refinement-V0.1%2BV0.2)?node-id=1226%3A0)
- [x] Related to JIRA ticket: GM-280(https://codecademy.atlassian.net/browse/GM-280)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.58-alpha.081494.0<br/>@codecademy/gamut-labs@19.15.1-alpha.081494.0<br/>@codecademy/gamut@34.13.1-alpha.081494.0<br/>@codecademy/styleguide@41.12.1-alpha.081494.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.59-alpha.c9e41f.0<br/>@codecademy/gamut-labs@19.15.2-alpha.c9e41f.0<br/>@codecademy/gamut@34.13.2-alpha.c9e41f.0<br/>@codecademy/styleguide@41.12.2-alpha.c9e41f.0<br/></pre>

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 1]
```diff
@@ -780,19 +780,23 @@ See the [`Radio`](/?path=/docs/atoms-forminputs-radio--radio) story for an examp
 ## Sections
 
 Our `GridForm`s optionally take an array of sections that have left and center-aligned variants.
-Each sections should have a title that correctly follows [heading ranks](https://usability.yale.edu/web-accessibility/articles/headings). You can set the text type of this title though the `as` prop on that section -- the default is `h2`.
+
+Each sections should have a title that correctly follows [heading ranks](https://usability.yale.edu/web-accessibility/articles/headings). You can set the text type of this title though the `as` prop on that section -- the default is `h2`. You can only set the `as` prop to a heading.
+
+You can set the rest of the [Text](https://gamut.codecademy.com/storybook/?path=/docs/typography-text--text) props for the section title the same way. Please check if your styling can be achieved with a [variant](https://gamut.codecademy.com/storybook/?path=/docs/typography-text--text#1-element-variants) before exploring other props.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:786]: What's the use case here? I'm super duper wary of giving people so much control over them... can it be a `Pick` of some allowed ones?

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60db85957e7098072f5055c7--gamut-preview.netlify.app>https://60db85957e7098072f5055c7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60db85957e7098072f5055c7>Deploy Logs</a></p>

**Review by katiezutter — APPROVED** [review]
Thank you thank you for getting this up so quickly!! ⚡️ Looks great on the Profile Edit page - here's the branch (preview environment with changes is not up yet): https://github.com/codecademy-engineering/Codecademy/pull/25557

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx (hunk 1 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:786]: @katiezutter's current PR is the best use-case -- I will experiment in restricting this to just `variants`. Does that make more sense ya think?

Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx]: Yeah that makes sense to me, thanks!

If you wanted to be _really_ snazzy you could restrict it to just the title variants. Unless that's an explicit list somewhere, you could probably write some fancy complex TypeScript type to restrict the string literal types... up to you 😄

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.59-alpha.477d88.0<br/>@codecademy/gamut-labs@19.15.2-alpha.477d88.0<br/>@codecademy/gamut@34.13.2-alpha.477d88.0<br/>@codecademy/styleguide@41.12.2-alpha.477d88.0<br/></pre>

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
What you suggested 😄

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60db8a57e1c17f0bfab8bb8a--gamut-preview.netlify.app>https://60db8a57e1c17f0bfab8bb8a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60db8a57e1c17f0bfab8bb8a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.59-alpha.02cb12.0<br/>@codecademy/gamut-labs@19.15.2-alpha.02cb12.0<br/>@codecademy/gamut@34.13.2-alpha.02cb12.0<br/>@codecademy/styleguide@41.12.2-alpha.02cb12.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60dc873b9e46d1374a72d639--gamut-preview.netlify.app>https://60dc873b9e46d1374a72d639--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60dc873b9e46d1374a72d639>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.59-alpha.f7139f.0<br/>@codecademy/gamut-labs@19.15.2-alpha.f7139f.0<br/>@codecademy/gamut@34.13.2-alpha.f7139f.0<br/>@codecademy/styleguide@41.12.2-alpha.f7139f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60dca155397c6b58b60e000f--gamut-preview.netlify.app>https://60dca155397c6b58b60e000f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60dca155397c6b58b60e000f>Deploy Logs</a></p>

**Review by sooinc — APPROVED** [review]
This is awesome!! Thank you for adding this. Tested in Katie's branch that it indeed throws a TS error if you aren't using title-size.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.61-alpha.248ede.0<br/>@codecademy/gamut-labs@19.15.4-alpha.248ede.0<br/>@codecademy/gamut@34.13.3-alpha.248ede.0<br/>@codecademy/styleguide@41.12.4-alpha.248ede.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60df66c674fefb4ac2200556--gamut-preview.netlify.app>https://60df66c674fefb4ac2200556--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60df66c674fefb4ac2200556>Deploy Logs</a></p>

**Pull request merged**